src/native-stack/utils/useBackPressSubscription.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import { BackHandler, NativeEventSubscription } from 'react-native';
+
+interface Args {
+  onBackPress: () => boolean;
+  isDisabled: boolean;
+}
+
+interface UseBackPressSubscription {
+  handleAttached: () => void;
+  handleDetached: () => void;
+  createSubscription: () => void;
+  clearSubscription: () => void;
+}
+
+/**
+ * This hook is an abstraction for keeping back press subscription
+ * logic in one place.
+ */
+export function useBackPressSubscription({
+  onBackPress,
+  isDisabled,
+}: Args): UseBackPressSubscription {
+  const [isActive, setIsActive] = React.useState(false);
+  const subscription = React.useRef<NativeEventSubscription | undefined>();
+
+  const clearSubscription = React.useCallback(
+    (shouldSetActive = true) => {
+      subscription.current?.remove();
+      subscription.current = undefined;
+      if (shouldSetActive) setIsActive(false);
+    },
+    [subscription.current]
+  );
+
+  const createSubscription = React.useCallback(() => {
+    if (!isDisabled) {
+      subscription.current?.remove();
+      subscription.current = BackHandler.addEventListener(
+        'hardwareBackPress',
+        onBackPress
+      );
+      setIsActive(true);
+    }
+  }, [subscription.current, isDisabled]);
+
+  const handleAttached = React.useCallback(() => {
+    if (isActive) {
+      createSubscription();
+    }
+  }, [createSubscription, isActive]);
+
+  const handleDetached = React.useCallback(() => {
+    clearSubscription(false);
+  }, [createSubscription, isActive]);
```

Comment:
`isActive` is not used here, so should it be in dependencies?

Reply:
It shouldn't be. Thanks for catching that